admin/sections/config/tools/openapi.tsx

```diff
@@ -0,0 +1,110 @@
+import { ToolConfigType } from "@/client/tool";
+import {
+  DEFAULT_OPENAPI_TOOL_CONFIG,
+  OpenAPIToolConfigType,
+} from "@/client/tools/openapi";
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { UseFormReturn } from "react-hook-form";
+
+// Todo: Use a separated form for OpenAPIConfig
+export const OpenAPIConfig = ({
+  form,
+  onSubmit,
+}: {
+  form: UseFormReturn<ToolConfigType>;
+  onSubmit: (tool_name: string, data: OpenAPIToolConfigType) => void;
+}) => {
+  return (
+    <>
+      <FormField
+        control={form.control}
+        name="openapi"
+        render={({ field }) => (
+          <FormItem
+            key={field.value.name}
+            className="flex flex-row items-center space-x-3 space-y-0"
+          >
+            <FormControl>
+              <Checkbox
+                checked={field.value.enabled ?? false}
+                onCheckedChange={(checked) => {
+                  // Submit the openapi config when the checkbox has changed and either:
+                  // 1. The checkbox is unchecked
+                  // 2. Or the checkbox is checked and the openapi_uri is valid
+                  field.onChange({
+                    ...field.value,
+                    enabled: checked,
+                  });
+                  const openAPIConfig = form.getValues().openapi;
+                  if (
+                    (checked &&
+                      openAPIConfig &&
+                      openAPIConfig?.config?.openapi_uri) ||
+                    !checked
+                  ) {
```

Comment:
can't we just run the validator from the schema and only if it's valid do the submit? 
then we might be able to use the same code as below

Reply:
I added this for a better UI. There're two things:
+ We don't need form validation if the user disabling the tool, otherwise the submit will fail if the validation has failed.
+ It's a bit weird that we show validation error instantly when user click enable the tool and haven't changed any config.

Reply:
>We don't need form validation if the user disabling the tool, otherwise the submit will fail if the validation has failed.
but shouldn't a disabled tool config also be valid?

Reply:
>It's a bit weird that we show validation error instantly when user click enable the tool and haven't changed any config.
that's a good point, we could count an empty string also as valid to solve this

Reply:
Let's keep it as is for now